gateway_test.go

```diff
@@ -1032,6 +1032,39 @@ func TestCacheAllSafeRequests(t *testing.T) {
 	}...)
 }
 
+func TestCachePostRequest(t *testing.T) {
+	ts := newTykTestServer()
+	defer ts.Close()
+	cache := storage.RedisCluster{KeyPrefix: "cache-"}
+	defer cache.DeleteScanMatch("*")
+
+	buildAndLoadAPI(func(spec *APISpec) {
+		spec.CacheOptions = apidef.CacheOptions{
+			CacheTimeout:         120,
+			EnableCache:          true,
+			CacheAllSafeRequests: false,
+		}
+
+		updateAPIVersion(spec, "v1", func(v *apidef.VersionInfo) {
+			json.Unmarshal([]byte(`[{
+						"method":"POST",
+						"path":"/",
+						"cache_key_regex":""
+					}
+                                ]`), &v.ExtendedPaths.AdvanceCacheConfig)
+		})
+
+		spec.Proxy.ListenPath = "/"
+	})
+
+	headerCache := map[string]string{"x-tyk-cached-response": "1"}
+
+	ts.Run(t, []test.TestCase{
+		{Method: "POST", Path: "/", HeadersNotMatch: headerCache, Delay: 10 * time.Millisecond},
```

Comment:
In this unit test are you actually sending a body with the post request?

Reply:
We should probably have a case testing the regex key against specific body.

Reply:
Fixed test cases.